src/mrpro/operators/_MatrixOp.py

```diff
@@ -0,0 +1,56 @@
+"""Matrix Multiplication Operator."""
+
+# Copyright 2024 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+
+import torch
+
+from mrpro.operators import LinearOperator
+
+
+class MatrixOp(LinearOperator):
+    """Simple Linear Operator that implements matrix multiplication."""
+
+    def __init__(self, matrix: torch.Tensor) -> None:
+        super().__init__()
+        self.register_buffer('matrix', matrix)
+
+    def forward(self, x) -> tuple[torch.Tensor]:
+        """Multiplication of input with matrix.
+
+        Parameters
+        ----------
+        x
+            tensor to be multiplied with matrix
+
+        Returns
+        -------
+            result of matrix-vector multiplication
+        """
+
+        return (self.matrix @ x,)
```

Comment:
Can we make this a bit more general with regards to batching?
As is, this will be of of limited real world use due to reshaping and batching required.
let my try out something....

Reply:
see the other PR where I requested your review